driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -594,6 +597,61 @@ public static Bson sample(final int size) {
         return new BsonDocument("$sample", new BsonDocument("size", new BsonInt32(size)));
     }
 
+    /**
+     * Creates a {@code $setWindowFields} pipeline stage, which allows using window operators.
+     * This stage partitions the input documents similarly to the {@link #group(Object, List) $group} pipeline stage,
+     * optionally sorts them, computes fields in the documents by computing window functions over {@linkplain Window windows} specified per
+     * function, and outputs the documents. The important difference from the {@code $group} pipeline stage is that
+     * documents belonging to the same partition or window are not folded into a single document.
+     *
+     * @param partitionBy Optional partitioning of data specified like {@code id} in {@link #group(Object, List)}.
+     *                    If {@code null}, then all documents belong to the same partition.
+     * @param sortBy Fields to sort by. The syntax is identical to {@code sort} in {@link #sort(Bson)} (see {@link Sorts}).
+     *               Sorting is required by certain functions and may be required by some windows (see {@link Windows} for more details).
+     *               Sorting is used only for the purpose of computing window functions and does not guarantee ordering
+     *               of the output documents.
+     * @param output A nonempty array of {@linkplain WindowedComputation windowed computations}.
+     * @param <TExpression> The {@code partitionBy} expression type.
+     * @return The {@code $setWindowFields} pipeline stage.
+     * @mongodb.driver.dochub core/window-functions-set-window-fields $setWindowFields
+     * @mongodb.server.release 5.0
+     * @since 4.3
+     */
+    @Beta
+    public static <TExpression> Bson setWindowFields(@Nullable final TExpression partitionBy, @Nullable final Bson sortBy,
```

Comment:
I think we could safely remove this overload.  It's not that helpful, and arguably misleading, since it's required that there be at least one windowed computation.  With group, by contrast, it's not required that there be any accumulators.  And with `List.of` it's quite easy to create a list of one (or more) items.

Reply:
I added this overload because I saw similar overloads for other aggregates, and anticipated a request to add it. Other than that, I do not see this overload being useful regardless of whether the `output` is allowed to be empty or not. So I would gladly remove the overload, but I went to read the syntax again, and it does say _"We will consider it a no-op rather than an error if no fields are provided."_ So I will remove the check `isTrueArgument("output must not be empty", !output.isEmpty())`.

Given the above, do you still want the overload to be removed or to stay?

Reply:
For consistency with group we should probably keep the overload then.